api/envoy/extensions/http/original_ip_detection/xff/v3/xff.proto

```diff
@@ -22,5 +26,49 @@ message XffConfig {
   // determining the origin client's IP address. The default is zero if this option
   // is not specified. See the documentation for
   // :ref:`config_http_conn_man_headers_x-forwarded-for` for more information.
+  //
+  // Only one of ``xff_num_trusted_hops`` and ``xff_trusted_cidrs`` should be set.
+  // If ``xff_trusted_cidrs`` is set, this field is ignored.
   uint32 xff_num_trusted_hops = 1;
+
+  // The `CIDR <https://tools.ietf.org/html/rfc4632>`_ ranges to trust when
+  // evaluating the remote IP address to determine the original client's IP address.
+  // This is used instead of
+  // :ref:`use_remote_address <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.use_remote_address>`.
+  // When the remote IP address matches a trusted CIDR and the
+  // :ref:`config_http_conn_man_headers_x-forwarded-for` header was sent, each entry
+  // in the ``x-forwarded-for`` header is evaluated from right to left and the first
+  // public non-trusted address is used as the original client address. If all
+  // addresses in ``x-forwarded-for`` are within the trusted list, the first (leftmost)
+  // entry is used.
+  //
+  // This is typically used when requests are proxied by a
+  // `CDN <https://en.wikipedia.org/wiki/Content_delivery_network>`_.
+  //
+  // Only one of ``xff_num_trusted_hops`` and ``xff_trusted_cidrs`` should be set.
+  // If set, takes precedence over ``xff_num_trusted_hops``.
+  XffTrustedCidrs xff_trusted_cidrs = 2;
+
+  // If set, Envoy will not append the remote address to the
+  // :ref:`config_http_conn_man_headers_x-forwarded-for` HTTP header.
+  //
+  // .. attention::
+  //
+  //   For proper proxy behaviour it is not recommended to set this option.
+  //   For backwards compatibility, if this option is unset and
+  //   ``xff_num_trusted_hops`` is set, it defaults to true. When
+  //   ``xff_trusted_cidrs`` is set, it defaults to false.
```

Comment:
Different default values only brings confusion. Let's only use a default value with `true`.

Reply:
Please see the related conversation below: https://github.com/envoyproxy/envoy/pull/31831#discussion_r1706979967

@alyssawilk seemed half inclined to go with this split default, but let me know if you still think we should really keep one default.

`true` is a bad default, to be honest, but I also don't want to break any existing users of `xff_num_trusted_hops`.

Reply:
From API's perspective, explicit and predictable behavior is always better. Different default values is always not recommend for our API.

Reply:
Got it. I've updated this to default to true.